src/explorer/Common/ValueCounts.cs

```diff
@@ -29,14 +33,28 @@ private ValueCounts()
 
         public double SuppressedCountRatio => TotalCount == 0 ? 1 : (double)SuppressedCount / TotalCount;
 
-        // Note the `SuppressedCount` can be seen as a proxy for the number of suppressed rows in the original dataset.
+        // Note the `SuppressedCount / MedianCount` can be seen as a proxy for the number of suppressed rows in the original dataset.
         // We can use this to estimate the proportion of unqiue values that have been suppressed. This may be a better
         // metric for estimating the cardinality of a column than the `SuppressedCountRatio`
-        public double SuppressedRowRatio => TotalRows == 0 ? 1 : (double)SuppressedCount / TotalRows;
+        public double SuppressedRowRatio => MedianCount == 0 || TotalRows == 0 ? 1 : (double)SuppressedCount / MedianCount / TotalRows;
+
+        public bool IsCategorical => SuppressedRowRatio < SuppressedRatioThreshold;
+
+        public static ValueCounts Compute(IList<CountableRow> rows)
+        {
+            var vc = rows.Aggregate(new ValueCounts(), AccumulateRow);
+            if (rows.Count != 0)
+            {
+                vc.MedianCount = rows.Count % 2 == 1 ?
+                        rows[(rows.Count - 1) / 2].Count :
```

Comment:
I think this line should be:
```suggestion
                        rows[(rows.Count + 1) / 2].Count :
```

Also, this algorithm only works if the `rows` are sorted by `Count` so I think the sorting should happen inside this method.

Reply:
I think that we can approximate the low-count threshold using the minimum count. What do you think, are there any problems with this solution?

Reply:
As mentioned by Paul earlier, the low-count threshold is always between 2 and 6. I think the simplest solution is just to approximate the low-count threshold as a constant `4`.

Reply:
After trying to fix this problem, I encountered some failing tests, and this made me think that we actually have two different problems here. 

The first problem is the one that we discuss here: how to decide if a column is categorical or not. After looking at the data from `GiveMeSomeCredit.Loans.MonthlyIncome`, which I would say that is not categorical (although there are some values that account for more than 1% of the rows), it seems that the condition with `SuppressedRowRatio` is better than the condition with `SuppressedCountRatio`. The condition with `SuppressedRowRatio` captures the fact that there are many values which are not part of a bucket.

We also have a second problem, where we want to generate sample data for text columns (issue #264). Since the algorithm with substrings is very imprecise, we can have a much more relaxed test for deciding when to chose from existing data as opposed to generating new sample data with substrings. I think that we can use a factor as low as 20%, that is: if more than 20% of the values are not suppressed, we can chose from the existing values, assuming also that the set of distinct non-suppressed rows is significantly bigger (e.g. an order of magnitude) than the number of samples we want.

Reply:
Argh just saw this comment now... This explains why travis isn't happy... 

Ok. I agree there are two problems here in that we need (a) a threshold for deciding if values are categorical and (b) a related threshold to decide when to use known string values. I also agree that the threshold for (b) can be much higher than the threshold for (a). I think that as long as the number of non-suppressed distinct string values is comparable to the number of samples we want to generate, we can sample from known values instead of synthesising.

Regarding the threshold to use for (a), I think we can do either of the following:
- fall back to using the `SuppressedRowRatio` with the normalizing factor of 2
  The problem with this is that is captures a measure of the proportion of categories that are suppressed, where as the `SuppressedCountRatio` captures suppression of the dataset as a whole. 
- use `SuppressedCountRatio`, but with a much lower threshold value, say 1%. In other words, 99% of the data should be captured in the category values. 

I think I prefer the second option. 

Also maybe we should limit the number of categories? For example, I think it should be clear that when there are 1000s of `integer` categories, the data is in fact better modeled as a distribution rather than a categorical set.

Reply:
I agree with you, using `SuppressedCountRatio` with a very low threshold seems to be a simpler and cleaner solution.
I also agree that we must limit the number of categories. Maybe the maximum number of categories should be somehow linked to the logarithm of total rows.

Reply:
Something related to the log of total rows sounds like a promising idea. Although I think for now constant limit is probably sufficient. 10 seems too low, 100 seems too high... Maybe 50 is a good compromise? 

I would suggest we publish the distinct values even if the categorical check returns `false` - this way we can still use the values even if there are many of them.